crates/red_knot/src/types.rs

```diff
@@ -349,7 +351,14 @@ impl std::fmt::Display for DisplayType<'_> {
             Type::Never => f.write_str("Never"),
             Type::Unknown => f.write_str("Unknown"),
             Type::Unbound => f.write_str("Unbound"),
-            Type::Class(class_id) => f.write_str(self.store.get_class(*class_id).name()),
+            // TODO functions and classes should display using a fully qualified name
+            Type::Class(class_id) => {
+                // TODO: how do we want to display types that are not expressible in annotations?
+                f.write_str("Exact[type[")?;
+                f.write_str(self.store.get_class(*class_id).name())?;
+                f.write_str("]]")
```

Comment:
Previous proposed spellings for "exactly the `str` class" (and not a subclass) have included:
- `Literal[str]`: https://mail.python.org/archives/list/typing-sig@python.org/message/FPQH5ZHFLHGB7PMWEUHTB4FOK5O6MN6Q/
- `TypeForm[str]`: https://mail.python.org/archives/list/typing-sig@python.org/message/O6AYFN7WN623W5EJ7LULSD4LBA2M3IOA/

I'd be happy with either `Exact[str]` or `Literal[str]`. I weakly prefer `Literal[str]`, because `Literal` elsewhere already specifies "exactly this value I'm specifying here, and no subtype or supertype of it"; and it also means that we can reuse an existing symbol rather than having to add a new special form (even if it's an internal-only special form)

Reply:
I like `Literal[...]` for this, I'll switch to that.